### PR TITLE
SONAR-8924 statistics on active rules in api/qualityprofiles/search

### PR DESCRIPTION
**Pull request by simonbrandhof**

### CODE REVIEW

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [MyBatis.java#L134](https://github.com/SonarSource/sonarqube/blob/f0d1c78affb04dffdfde08272c07e69539c1a186/server/sonar-db-dao/src/main/java/org/sonar/db/MyBatis.java#L134): This method has 101 lines, which is greater than the 100 lines authorized. Split it into smaller methods. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS138)
1. ![INFO][INFO] [QualityProfileDao.java#L158](https://github.com/SonarSource/sonarqube/blob/f0d1c78affb04dffdfde08272c07e69539c1a186/server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java#L158): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [QualityProfileDao.java#L171](https://github.com/SonarSource/sonarqube/blob/f0d1c78affb04dffdfde08272c07e69539c1a186/server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java#L171): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [QualityProfileDao.java#L183](https://github.com/SonarSource/sonarqube/blob/f0d1c78affb04dffdfde08272c07e69539c1a186/server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java#L183): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: server/sonar-server/src/main/java/org/sonar/server/rule/index/RuleIndexDefinition.java [hunk 1]
```diff
@@ -65,6 +65,7 @@
 
   // Active rule fields
   public static final IndexType INDEX_TYPE_ACTIVE_RULE = new IndexType(INDEX, "activeRule");
+  public static final String FIELD_ACTIVE_ORGANIZATION_UUID = "orgUuid";
```
Comment (bartfastiel) [inline server/sonar-server/src/main/java/org/sonar/server/rule/index/RuleIndexDefinition.java]: I would have preferred "organizationUuid" out of personal preference.

File: server/sonar-server/src/test/java/org/sonar/server/qualityprofile/index/ActiveRuleResultSetIteratorTest.java [hunk 2]
```diff
@@ -136,16 +138,7 @@ public void select_after_date() {
...
-  }
-
-  private enum DocToKey implements Function<ActiveRuleDoc, ActiveRuleKey> {
-    INSTANCE;
-
-    @Override
-    public ActiveRuleKey apply(@Nonnull ActiveRuleDoc doc) {
-      return doc.key();
-    }
+    return Maps.uniqueIndex(it, ActiveRuleDoc::key);
```
Comment (bartfastiel) [inline server/sonar-server/src/test/java/org/sonar/server/qualityprofile/index/ActiveRuleResultSetIteratorTest.java]: 👍

File: server/sonar-db-dao/src/main/resources/org/sonar/db/qualityprofile/ActiveRuleMapper.xml [hunk 3]
```diff
@@ -164,5 +164,28 @@
     from active_rule_parameters p
   </select>
 
+  <select id="countActiveRulesByProfileKey" resultType="KeyLongValue" parameterType="map">
+    select p.kee as "key", count(ar.id) as "value" from active_rules ar
```
Comment (bartfastiel) [inline server/sonar-db-dao/src/main/resources/org/sonar/db/qualityprofile/ActiveRuleMapper.xml]: I would prefer "from" to be on a new line.

File: server/sonar-db-dao/src/main/resources/org/sonar/db/qualityprofile/ActiveRuleMapper.xml [hunk 4]
```diff
@@ -164,5 +164,28 @@
...
+    inner join rules_profiles p on p.id = ar.profile_id
+    where p.organization_uuid = #{organizationUuid, jdbcType=VARCHAR}
+    group by p.kee
+  </select>
+
+  <select id="countActiveRulesForRuleStatusByProfileKey" resultType="KeyLongValue" parameterType="map">
+    select p.kee as "key", count(ar.id) as "value" from active_rules ar
```
Comment (bartfastiel) [inline server/sonar-db-dao/src/main/resources/org/sonar/db/qualityprofile/ActiveRuleMapper.xml]: I would prefer "from" to be on a new line.

File: server/sonar-db-dao/src/main/resources/org/sonar/db/qualityprofile/ActiveRuleMapper.xml [hunk 5]
```diff
@@ -164,5 +164,28 @@
...
+    inner join rules_profiles p on p.id = ar.profile_id
+    inner join rules r on r.id = ar.rule_id
+    where p.organization_uuid = #{organizationUuid, jdbcType=VARCHAR}
+    and r.status = #{ruleStatus, jdbcType=VARCHAR}
+    group by p.kee
+  </select>
+
+  <select id="countActiveRulesForInheritanceByProfileKey" resultType="KeyLongValue" parameterType="map">
+    select p.kee as "key", count(ar.id) as "value" from active_rules ar
```
Comment (bartfastiel) [inline server/sonar-db-dao/src/main/resources/org/sonar/db/qualityprofile/ActiveRuleMapper.xml]: I would prefer "from" to be on a new line.

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/index/ActiveRuleIndex.java [hunk 6]
```diff
@@ -1,146 +0,0 @@
...
-      } else if (aggregation instanceof InternalValueCount) {
-        InternalValueCount count = (InternalValueCount) aggregation;
-        FacetValue facetValue = new FacetValue(count.getName(), count.getValue());
-        stats.put(count.getName(), facetValue);
-      }
-    }
-    return stats;
-  }
-
-}
```
Comment (bartfastiel) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/index/ActiveRuleIndex.java]: I ❤️ this 💧

File: server/sonar-server/src/test/java/org/sonar/server/qualityprofile/QProfileServiceMediumTest.java [hunk 7]
```diff
@@ -1,191 +0,0 @@
...
-      org.sonar.api.rules.ActiveRule activeRule = rulesProfile.activateRule(rule, RulePriority.CRITICAL);
-      activeRule.setParameter("acceptWhitespace", "true");
-      return rulesProfile;
-    }
-  }
-
-  private void logInAsQProfileAdministrator() {
-    userSessionRule.logIn().addPermission(ADMINISTER_QUALITY_PROFILES, tester.get(DefaultOrganizationProvider.class).get().getUuid());
-  }
-}
```
Comment (bartfastiel) [inline server/sonar-server/src/test/java/org/sonar/server/qualityprofile/QProfileServiceMediumTest.java]: -1 medium test...

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/index/ActiveRuleIndex.java [hunk 8]
```diff
@@ -63,8 +63,8 @@ public ActiveRuleIndex(EsClient client) {
   public Map<String, Long> countAllByQualityProfileKey() {
     return countByField(FIELD_ACTIVE_RULE_PROFILE_KEY,
       QueryBuilders.hasParentQuery(INDEX_TYPE_RULE.getType(),
-        QueryBuilders.boolQuery().mustNot(
-          QueryBuilders.termQuery(FIELD_RULE_STATUS, RuleStatus.REMOVED.name()))));
+        QueryBuilders.boolQuery()
+          .mustNot(QueryBuilders.termQuery(FIELD_RULE_STATUS, RuleStatus.REMOVED.name()))));
```
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/index/ActiveRuleIndex.java]: ?

**Comment by simonbrandhof** [comment]
Feedback applied.

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [MyBatis.java#L134](https://github.com/SonarSource/sonarqube/blob/02d74264d020cbfc67c11a3d7d51787f382f2071/server/sonar-db-dao/src/main/java/org/sonar/db/MyBatis.java#L134): This method has 101 lines, which is greater than the 100 lines authorized. Split it into smaller methods. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS138)
1. ![INFO][INFO] [QualityProfileDao.java#L158](https://github.com/SonarSource/sonarqube/blob/02d74264d020cbfc67c11a3d7d51787f382f2071/server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java#L158): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [QualityProfileDao.java#L171](https://github.com/SonarSource/sonarqube/blob/02d74264d020cbfc67c11a3d7d51787f382f2071/server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java#L171): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [QualityProfileDao.java#L183](https://github.com/SonarSource/sonarqube/blob/02d74264d020cbfc67c11a3d7d51787f382f2071/server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java#L183): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Pull request closed without merging**